bus-mapping/src/evm/opcodes/ternop.rs

```diff
@@ -0,0 +1,129 @@
+use super::Opcode;
+use crate::circuit_input_builder::CircuitInputStateRef;
+use crate::eth_types::GethExecStep;
+use crate::{
+    operation::{StackOp, RW},
+    Error,
+};
+
+/// Placeholder structure used to implement [`Opcode`] trait over it corresponding to all the
+/// ternary operations: take three words and return one.
+#[derive(Debug, Copy, Clone)]
+pub(crate) struct TernaryOpcode;
+
+impl Opcode for TernaryOpcode {
```

Comment:
Is it possible to have something like `StackOpOnlyOpcode<const N: usize>` where `N` tells how many word to pop? Then `BinaryOpcode` and `UnaryOpcode` are just `StackOnlyOpcode<2>` and `StackOnlyOpcode<1>`.

Reply:
That's not a bad one. It makes everything more generic. But would like to see @ed255 perspective!

Reply:
> Is it possible to have something like `StackOpOnlyOpcode<const N: usize>` where `N` tells how many word to pop? Then `BinaryOpcode` and `UnaryOpcode` are just `StackOnlyOpcode<2>` and `StackOnlyOpcode<1>`.

That's a great suggestion!  I've applied it :)